File: projects/10/ruby/Ostrich.rb
# Ostrich.rb | Defines the Ostrich class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Sinai Park
# Date: APril 24 2020
####################################################

require './WalkingBird.rb'

class Ostrich < WalkingBird

######################################################
# call returns the sound that a Ostrich type makes       #
# Recieve: NONE                                      #
# Return: Snork!, a String                           #
######################################################
def call
	"Snork!"
end

end

File: projects/10/java/Ostrich.java
/* Ostrich.java provides Ostrich subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Sinai Park
 * Date: April 24 2020
 ******************************************************/

public class Ostrich extends WalkingBird {
    
     /* explicit constructor
      * Receive: name, a String 
      * PostCond: myName == name.
      */
    public Ostrich(String name)
    {
        super(name);
    }
    
    
     /* A Ostrich's Call
      * Return: a Ostrich-call ("Snork!").
      */
    public String call()
    {
        return "Snork!";
    }
    
    }
    
    